base/irrationals.jl

```diff
@@ -221,6 +221,15 @@ end
 log(::Irrational{:e}) = 1 # use 1 to correctly promote expressions like log(x)/log(e)
 log(::Irrational{:e}, x::Number) = log(x)
 
+# These methods should go to special/trig.jl, but Irrational is not yet defined when the
+# file is loaded.
+sinpi(x::Irrational) = sinpi(float(x))
+cospi(x::Irrational) = cospi(float(x))
+sinc(x::Irrational) = sinc(float(x))
+cosc(x::Irrational) = cosc(float(x))
+sinc(z::Complex{<:Irrational}) = sinc(float(z))
+cosc(z::Complex{<:Irrational}) = cosc(float(z))
```

Comment:
In this regard, an alternative is to change the signature of `sinpi(x::T) where T<:Real` to `sinpi(x::T) where T<:Union{Integer, Rationa}` and then define
```julia
sinpi(x::T) where T<:Real = sinpi(float(x))
```
so this wouldn't limited to irrationals, but would work with all other custom subtype of `Real`.  What do you think?